coffeemocha ip

coffeemocha opened the pull request:
# GodBot

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://www.brainyquote.com/quotes/david_allen_227416))

GodBot frees your mortal mind from the burden of remembering tasks. It's:
- text-based
- easy to summon and command
- **FAST** _DIVINELY_ FAST to use

All you need to do is:
1. [download it from here](https://github.com)
2. invoke it with a double-click
3. offer your tasks as tribute
4. let GodBot manage your mortal tasks for you ⚡

And it is **FREE** (as all divine gifts should be)!

## Features:
- [x] Managing mortal tasks
- [ ] Managing deadlines (awaiting divine intervention)
- [ ] Sending celestial reminders (coming soon)

---

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Commit:
Changed Duke to GodBot and additionally added .class in gitignore

Commit:
level-1 tag added personality and echo functionality

Commit:
Implemented level 2 and also made stuff into functions

Commit:
Implemented level 3, i added a new task class and stuff

Commit:
Used heirarchy to implement the different todo deadline and events. so new classes for them

Commit:
added some basic input.txt and edited the script to point to godbot

Commit:
Added 2 error checkings, one for invalid command and another for inproper index.

Commit:
Added delete functionality with error checking.:

Commit:
Implemented Level-7 saving and loading files

Commit:
Merging level 7 branch

Commit:
Implemented Level-8 date/time

Commit:
Merge Level 8 date/time

Commit:
A-MoreOOP implemented

Commit:
Merge branch 'branch-A-MoreOOP'

Commit:
Refactored to packages

coffeemocha committed:
Merged A-Packages into master

coffeemocha committed:
Merged add-gradle-support into master

coffeemocha committed:
Implemented Gradle build setup

coffeemocha committed:
Added 2 JUnit tests Task and Parser

coffeemocha committed:
docs: Add JavaDoc comments to public classes and methods

coffeemocha committed:
style: Apply coding standard adjustments to existing .java files

coffeemocha committed:
Implement find functionality to search tasks by keyword

coffeemocha committed:
Merge tag 'A-JavaDoc'

coffeemocha committed:
Merge remote-tracking branch 'origin/A-CodingStandard'

coffeemocha committed:
Merge remote-tracking branch 'origin/Level-9-Find'

limshaoqi commented on src/main/java/godbot/GodBot.java:
```diff
@@ -0,0 +1,67 @@
+    private TaskList tasks;
+    private Storage storage;
+    private Ui ui;
+    private static final String DEFAULT_FILE_PATH = "./data/godbot.txt";
```
The constant DEFAULT_FILE_PATH is well-named, using uppercase letters with underscores to separate words. This follows convention for constants. 
Good job <3

coffeemocha replied:
We love constants over here! Thanks <3

limshaoqi commented on src/main/java/godbot/GodBot.java:
```diff
@@ -0,0 +1,67 @@
+     *
+     * @param filePath The path to the file used for storing task data.
+     */
+    public GodBot(String filePath) {
```
The class name GodBot is clear and follows PascalCase convention. Nice skrtt!

coffeemocha replied:
Thanks!! I made sure to follow PascalCase throughout!

coffeemocha committed:
Implement GUI for GodBot using JavaFX

coffeemocha committed:
Level 10: Recommit changes with correct JavaFX Additions

coffeemocha committed:
Add assertions to improve code robustness

Assertions were missing from critical parts of the code, making it harder to catch incorrect states and invalid inputs early. This could lead to unexpected behavior during execution.

Adding assertions in key locations helps ensure assumptions about variables hold true, reducing the likelihood of silent failures. This also improves maintainability by explicitly stating expected conditions.

Let's:
- Add assertions to validate task indices before marking or unmarking tasks.
- Ensure file paths are not null when loading and saving task data.
- Verify command arguments are non-empty when parsing user input.

Assertions will help prevent subtle bugs and make the code more reliable.

coffeemocha committed:
Refactor variable names in GodBot.java for clarity

Renamed variables in the GodBot class to adhere to Java naming conventions
and improve code readability. This change enhances maintainability without
altering functionality.

coffeemocha committed:
Merge pull request #1 from coffeemocha/branch-A-Assertions

Add assertions to improve code robustness

coffeemocha committed:
Merge branch 'master' into branch-A-CodeQuality

coffeemocha committed:
Merge pull request #2 from coffeemocha/branch-A-CodeQuality

Refactor variable names in GodBot.java for clarity

coffeemocha committed:
Merge branch 'branch-Level-10' into BCD-Extension

coffeemocha committed:
feat: Add reminder functionality in Parser

Implemented a reminder functionality in the Parser to notify
users about upcoming tasks. The reminder command allows users
to check for tasks with approaching deadlines.

- Added support for the 'remind' command in Parser.
- Integrated reminder logic to fetch tasks with upcoming deadlines.
- Ensured the reminder system works without altering existing UI
  or storage mechanisms.

This enhancement helps users keep track of important tasks
without modifying other components of the system.

coffeemocha committed:
feat(gui): integrate GUI responses and close window on "bye"

- Redirected GodBot output from CLI to GUI using DialogBox.
- Ensured welcome message appears in the GUI on startup.
- Fixed user message alignment to the right.
- Implemented `Platform.exit()` to close the window when "bye" is entered.
- Refactored `Parser.processCommand()` to return responses instead of using `Ui.showMessage()`.

This commit fully transitions GodBot from a CLI-based chatbot to a GUI-integrated chatbot.

coffeemocha committed:
Merge branch 'BCD-Extension' into JavaFX-4

coffeemocha committed:
Merge pull request #5 from coffeemocha/branch-Level-10

Add JavaFX

coffeemocha committed:
Remove tabs

coffeemocha committed:
Remove tabs

coffeemocha committed:
Resolve merge conflicts

coffeemocha committed:
bug fix: Edit wrong variable name

coffeemocha committed:
Add mark functionality

coffeemocha committed:
Add ui.png

coffeemocha committed:
Add Ui.png

coffeemocha committed:
Delete ui.png

coffeemocha committed:
Add Ui.png again

coffeemocha committed:
Update README.md

coffeemocha committed:
Update README.md

coffeemocha committed:
Update README.md

coffeemocha committed:
Update README.md

coffeemocha committed:
bug fix: Remove unnecessary |

coffeemocha committed:
Merge branch 'master' of https://github.com/coffeemocha/ip

coffeemocha committed:
Update README.md

coffeemocha committed:
Update README.md

coffeemocha committed:
Merge branch 'master' of https://github.com/coffeemocha/ip

coffeemocha committed:
Update README.md

coffeemocha committed:
bug fix: Add time handling
